src/DfsStrategy.cpp

```diff
@@ -1,12 +1,77 @@
 #include "DfsStrategy.hpp"
```

Comment:
Both strategies looks just like copy paste; maybe that code could be reused somehow w/o loosing performance.